src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OIDCLoginBackingBean.java

```diff
@@ -0,0 +1,169 @@
+                            List.of(emailAddress));
+                    logger.log(Level.INFO, "redirect to first login: " + userRecordIdentifier);
+                    newAccountPage.setNewUser(userRecord);
+                    Faces.redirect("/oauth2/firstLogin.xhtml");
+                }
+            } else {
+                dvUser = userService.updateLastLogin(dvUser);
+                session.setUser(dvUser);
+                Faces.redirect("/");
+            }
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "Setting user failed: " + e.getMessage());
+        }
+    }
+
+    public UserRecordIdentifier getUserRecordIdentifier() {
+        try {
+            final String subject = openIdContext.getSubject();
+            final OIDCAuthProvider provider = getProvider();
+            return new UserRecordIdentifier(provider.getId(), subject);
+        } catch (final Exception ignore) {
+            return null;
+        }
+    }
+
+    private String getVerifiedEmail() {
+        try {
+            if (openIdContext.getAccessToken().isExpired()) {
+                return null;
+            }
+            final Object emailVerifiedObject = openIdContext.getClaimsJson().get(OpenIdConstant.EMAIL_VERIFIED);
+            final boolean emailVerified;
+            if (emailVerifiedObject instanceof JsonValue) {
+                final JsonValue v = (JsonValue) emailVerifiedObject;
+                emailVerified = JsonValue.TRUE.equals(v)
+                        || (JsonValue.ValueType.STRING.equals(v.getValueType())
+                                && Boolean.getBoolean(((JsonString) v).getString()));
+            } else {
+                emailVerified = false;
+            }
+            if (!emailVerified) {
+                logger.log(Level.SEVERE,
+                        "email not verified: " + openIdContext.getClaimsJson().get(OpenIdConstant.EMAIL));
+                return null;
+            }
+            return openIdContext.getClaims().getEmail().orElse(null);
+        } catch (final Exception ignore) {
+            return null;
+        }
+    }
+
+    private OIDCAuthProvider getProvider() {
+        final String issuerEndpointURL = openIdContext.getAccessToken().getJwtClaims()
+                .getStringClaim(OpenIdConstant.ISSUER_IDENTIFIER)
+                .orElse(null);
+        if (issuerEndpointURL == null) {
+            logger.log(Level.SEVERE,
+                    "Issuer URL (iss) not found in " + openIdContext.getAccessToken().getJwtClaims().toString());
+            return null;
+        }
+        List<OIDCAuthProvider> providers = authenticationSvc.getAuthenticationProviderIdsOfType(OIDCAuthProvider.class)
+                .stream()
+                .map(providerId -> (OIDCAuthProvider) authenticationSvc.getAuthenticationProvider(providerId))
+                .filter(provider -> issuerEndpointURL.equals(provider.getIssuerEndpointURL()))
```

Comment:
Are we sure these values are equal? Does the issuer URL have to be the full qualified one or could it be just the "top" URL from where you can access the `.well-known` endpoint?